src/rpc/blockchain.cpp

```diff
@@ -1069,12 +1069,9 @@ static RPCHelpMan pruneblockchain()
     }
 
     PruneBlockFilesManual(active_chainstate, height);
-    const CBlockIndex* block = active_chain.Tip();
-    CHECK_NONFATAL(block);
-    while (block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA)) {
-        block = block->pprev;
-    }
-    return uint64_t(block->nHeight);
+    const CBlockIndex* last_block = GetLastPrunedBlock(*active_chain.Tip());
```

Comment:
In commit "refactor: Introduce GetLastPruneBlock helper function" (292d508e956bfe5b05e6ee685415063c2a0f04af)

Note: Here and in the other location below there's no more CHECK_NONFATAL so there would be segfault instead of an exception if the tip was null. Doesn't seem like a problem, but if were for some reason, GetLastPrunedBlock argument could be a pointer instead of a reference and the check could be added there.

Reply:
True, I changed the function to take a pointer and added a CHECK_NONFATAL to be more consistent with the previous code.